ansible_collections/arista/avd/plugins/module_utils/eos_designs_facts.py

```
@@ -160,6 +167,31 @@ def default_evpn_role(self):
         """
         return get(self._node_type_key_data, "default_evpn_role", default="none")
 
+    @cached_property
+    def default_interfaces(self):
+        default_interfaces = get(self._hostvars, "default_interfaces", default=[])
+
+        if self.platform is None:
```

Note to other reviewers - we have some minimal device configs in molecule/eos_designs_unit_tests that don't have a platform set that were causing this to fail.  'platform' is a mandatory setting however, so I don't know if we should bail out with an error here or not or just silently proceed.

---

I don't think `platform` is mandatory at all. It should just match the `default` platform_setting or `default_interfaces` in this case. So you could set it to `"default"` if the platform is not there and that will match the default default_interface during the first run-through.

---

This is fixed as discussed in latest commit.